### PR TITLE
Update Series.value_counts() to handle dropna argument and handle NaN/None values

### PR DESCRIPTION
**Pull request by kozlov-alexey**

### CODE REVIEW

File: sdc/datatypes/hpat_pandas_series_functions.py [hunk 1]
```diff
@@ -445,65 +448,131 @@ def hpat_pandas_series_value_counts(self, normalize=False, sort=True, ascending=
...
+                # positive and negative zeros, hence we count zero values separately and store
+                # as a key the first zero value found in the Series
+                if not value:
+                    zero_counts += 1
+                    if not is_zero_found:
+                        zero_value = value
+                        is_zero_found = True
+                    continue
+
+                if value in value_counts_dict:
```
Comment (AlexanderKalistratov) [inline sdc/datatypes/hpat_pandas_series_functions.py]: consider:
```python
value_counts_dict[value] = value_counts_dict.get(value, 0) + 1
```

File: sdc/datatypes/hpat_pandas_series_functions.py [hunk 2]
```diff
@@ -445,65 +448,131 @@ def hpat_pandas_series_value_counts(self, normalize=False, sort=True, ascending=
...
+                    if not dropna:
+                        nan_counts += 1
+                    continue
+
                 if value in value_counts_dict:
                     value_counts_dict[value] += 1
                 else:
                     value_counts_dict[value] = 1
 
+                if (not empty_str_key_exist and value == ''):
```
Comment (AlexanderKalistratov) [inline sdc/datatypes/hpat_pandas_series_functions.py]: consider:
```python
empty_str_key_exist = empty_str_key_exist or (value == '')
```

File: sdc/datatypes/hpat_pandas_series_functions.py (hunk 1 above)
Comment (kozlov-alexey) [reply sdc/datatypes/hpat_pandas_series_functions.py]: @AlexanderKalistratov Thank you! Great that it's supported in nopython mode.

File: sdc/datatypes/hpat_pandas_series_functions.py (hunk 2 above)
Comment (kozlov-alexey) [reply sdc/datatypes/hpat_pandas_series_functions.py]: Fixed

**Comment by AlexanderKalistratov** [comment]
@kozlov-alexey please merge it ASAP! 😄

File: sdc/datatypes/hpat_pandas_series_functions.py [hunk 3]
```diff
@@ -445,66 +448,118 @@ def hpat_pandas_series_value_counts(self, normalize=False, sort=True, ascending=
...
+                value_counts_dict[value] = value_counts_dict.get(value, 0) + 1
+
+            need_add_nan_count = not dropna and nan_counts
 
-            # TODO: workaround, keys() result can not be casted to array type
-            # TODO: use list comprehension instead or self.unique()
-            unique_values = [key for key in value_counts_dict]
-            unique_values_len = len(unique_values)
+            values = [key for key in value_counts_dict]
+            counts_as_list = [value_counts_dict[key] for key in value_counts_dict.keys()]
```
Comment (densmirn) [inline sdc/datatypes/hpat_pandas_series_functions.py]: Are we able to use method `values()` directly?
`[val for val in value_counts_dict.values()]`

File: sdc/datatypes/hpat_pandas_series_functions.py [hunk 4]
```diff
@@ -445,66 +448,118 @@ def hpat_pandas_series_value_counts(self, normalize=False, sort=True, ascending=
...
+            # allocate the result index as a StringArray and copy values to it
+            index_string_lengths = numpy.asarray([len(s) for s in values_sorted_by_count])
+            index_total_chars = numpy.sum(index_string_lengths)
+            result_index = pre_alloc_string_array(len(values_sorted_by_count), index_total_chars)
+            cp_str_list_to_array(result_index, values_sorted_by_count)
+
+            if need_add_nan_count:
+                # set null bit for StringArray element corresponding to NaN element (was added as last in values)
+                index_previous_nan_pos = values_len - 1
+                for i in numpy.arange(values_len):
```
Comment (densmirn) [inline sdc/datatypes/hpat_pandas_series_functions.py]: Shouldn't we use `numba.prange` here?

File: sdc/datatypes/hpat_pandas_series_functions.py (hunk 3 above)
Comment (kozlov-alexey) [reply sdc/datatypes/hpat_pandas_series_functions.py]: Yes, but for some strange reason it produces different results. I'll investigate that separately.

**Comment by kozlov-alexey** [comment]
LGTM

**Pull request merged**